torchvision/datasets/utils.py

```diff
@@ -244,12 +249,17 @@ def _get_confirm_token(response: "requests.models.Response") -> Optional[str]:
 
 
 def _save_response_content(
-    response: "requests.models.Response", destination: str, chunk_size: int = 32768,  # type: ignore[name-defined]
+    response_gen: Iterator[bytes], destination: str, first_chunk: bytes,  # type: ignore[name-defined]
```

Comment:
IIUC, `first_chunk` is the first relevant element of the now-partly-consumed `response_gen` generator?

To avoid the duplicated code, maybe it would it be better to pass a single generator by calling the function with something like 
```py
_save_response_content(itertools.chain([first_chunk], response_content_generator), ... )
```
?

Reply:
You got that correct and it does indeed simplify the code, realised your suggestion in 581bde3597616946503a97f59bb4b2369dc3d09d